config/routes.rb

```diff
@@ -64,7 +64,7 @@
       jsonapi_resources :studies
       jsonapi_resources :submission_pools
       jsonapi_resources :submission_templates
-      jsonapi_resources :submissions
+      jsonapi_resources :submissions, except: %i[update]
```

Comment:
Curious about that restriction and also state being read only.
Wondering if we need to be able to directly update submission state. Or if's all done indirectly through requests...

Reply:
This pull request takes an endpoint that could not previously be created or updated (i.e. it was `immutable`) and adds a create POST endpoint while still leaving the update PATCH endpoint turned off.  If we need to update state, it's not needed so far, but could be enabled carefully in future.  `immutable` in the resource effectively adds `, only: %i[show]` to the route without explicitly writing it.  I think it's a poor choice by the gem creators and they should expect you to just add that annotation to the routes file instead so it's explicit.  As it is, you cannot tell from the routes alone which are immutable and which are full access.